README.md

```diff
@@ -1,368 +1,257 @@
+    Create react on rails files for install generator.
 ```
 
+We have a repo showing the results of running the generator with various combinations of options, each combination on its own branch:
 
-#### Sample webpack.server.rails.config.js (ONLY for server rendering)
-Be sure to check out the latest example version of [client/webpack.server.rails.config.js](https://github.com/shakacode/react-webpack-rails-tutorial/blob/master/client/webpack.server.rails.config.js).
++ [basic (`rails generate react_on_rails:install`)](https://github.com/shakacode/react_on_rails-generator-results/tree/basic)
++ [basic-server-rendering (`rails generate react_on_rails:install --server-rendering`)](https://github.com/shakacode/react_on_rails-generator-results/tree/basic-server-rendering)
++ [redux (`rails generate react_on_rails:install --redux`)](https://github.com/shakacode/react_on_rails-generator-results/tree/redux)
++ [redux-server-rendering (`rails generate react_on_rails:install --redux --server-rendering`)](https://github.com/shakacode/react_on_rails-generator-results/tree/redux-server-rendering)
 
-```javascript
-  // Common webpack configuration for server bundle
-
-module.exports = {
-
-  // the project dir
-  context: __dirname,
-  entry: ['./app/startup/serverGlobals'],
-  output: {
-    filename: 'server-bundle.js',
-    path: '../app/assets/javascripts/generated',
-
-    // CRITICAL to set libraryTarget: 'this' for enabling Rails to find the exposed modules IF you
-    //   use the "expose" webpackfunctionality. See startup/serverGlobals.jsx.
-    // NOTE: This is NOT necessary if you use the syntax of global.MyComponent = MyComponent syntax.
-    // See http://webpack.github.io/docs/configuration.html#externals for documentation of this option
-    //libraryTarget: 'this',
-  },
-  resolve: {
-    extensions: ['', '.webpack.js', '.web.js', '.js', '.jsx', 'config.js'],
-  },
-  module: {
-    loaders: [
-      {test: /\.jsx?$/, loader: 'babel-loader', exclude: /node_modules/},
-
-      // React is necessary for the client rendering:
-      { test: require.resolve('react'), loader: 'expose?React' },
-      { test: require.resolve('react-dom/server'), loader: 'expose?ReactDOMServer' },
-    ],
-  },
-};
-```
+## How it Works
+The generator installs your webpack files in the `client` folder. You then use webpack to compile your code and output the bundled results to `app/assets/javascripts/generated`, which are then loaded by sprockets. These generated bundle files have been added to your `.gitignore` for your convenience.
 
-## What Happens?
+In most cases, you should then use the provided helper method to render the React component from your Rails views. In some cases, such as when SEO is vital or many users will not have JavaScript enabled, you can pass the `--server-rendering` option to the generator to configure your application for server-side rendering. Your JavaScript can then be first rendered on the server and passed to the client as HTML.
```

Comment:
You're discussing with server rendering or without (just client rendering).



This could be more clear.

react_on_rails supports server rendering which means that your JavaScript can then be first rendered on the server and passed to the client as HTML.

In most cases, you should use the render_component helper option `prerender: false` and do only client side rendering. In some cases, such as when SEO is vital or when many users will not have JavaScript enabled, you can use the render_component helper `prerender: true` do do server rendering. There's a bit more than just this.

The best way to see what's required is to use the `--server-rendering` option to the generator to configure your application for server-side rendering.